extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -292,20 +361,189 @@ export class AmpStoryQuiz extends AMP.BaseElement {
     );
   }
 
+  /**
+   * @param {Element} selectedOption
+   * @private
+   */
+  updateQuizToPostSelectionState_(selectedOption) {
+    this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+    selectedOption.classList.add('i-amphtml-story-quiz-option-selected');
+  }
+
   /**
    * Triggers changes to quiz state on response interaction.
    *
    * @param {!Element} optionEl
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.triggerAnalytics_(optionEl);
+    this.responseDataPromise_.then(() => {
+      if (this.hasUserSelection_) {
+        return;
+      }
 
-    this.mutateElement(() => {
-      optionEl.classList.add('i-amphtml-story-quiz-option-selected');
-      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+      this.triggerAnalytics_(optionEl);
+
+      this.mutateElement(() => {
+        this.updateQuizToPostSelectionState_(optionEl);
+
+        this.hasUserSelection_ = true;
```

Comment:
This shouldn't be in the `mutateElement` as it runs asynchronously. This could lead to bugs like calling the `updateReactionData_()` method multiple times, as the early return would not work right away.

Reply:
Done!